### PR TITLE
upgrade rustdoc's `pulldown-cmark` to 0.5.2

### LINKED ISSUE #60482: nightly `cargo doc` panics in font8x8 crate
`cargo +nightly doc` fails for the [font8x8](https://crates.io/crates/font8x8) crate. Stable works fine.

```
$ cargo +nightly doc
 Documenting font8x8 v0.2.4 (/home/richard/src/font8x8-rs)
thread 'rustc' panicked at 'unexpected node in tree: Header(2)', /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:1019:29
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.36.0-nightly (9b67bd42b 2019-05-01) running on x86_64-unknown-linux-gnu

error: Could not document `font8x8`.

Caused by:
  process didn't exit successfully: `rustdoc --crate-name font8x8 src/lib.rs --color always -o /home/richard/src/font8x8-rs/target/doc --cfg 'feature="default"' --cfg 'feature="std"' --cfg 'feature="unicode"' -L dependency=/home/richard/src/font8x8-rs/target/debug/deps` (exit code: 1)
```

## Meta

Backtrace:

```
$ RUST_BACKTRACE=full cargo +nightly doc
 Documenting font8x8 v0.2.4 (/home/richard/src/font8x8-rs)
thread 'rustc' panicked at 'unexpected node in tree: Header(2)', /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:1019:29
stack backtrace:
   0:     0x7f89f6e27e23 - std::sys::unix::backtrace::tracing::imp::unwind_backtrace::hdee1923a5e5a6475
                               at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1:     0x7f89f6e1fcab - std::sys_common::backtrace::_print::hfc77fdbc72a5d365
                               at src/libstd/sys_common/backtrace.rs:71
   2:     0x7f89f6e24086 - std::panicking::default_hook::{{closure}}::h051f6b831cd488ff
                               at src/libstd/sys_common/backtrace.rs:59
                               at src/libstd/panicking.rs:197
   3:     0x7f89f6e23e19 - std::panicking::default_hook::h4b88024c1e3f38ad
                               at src/libstd/panicking.rs:211
   4:     0x7f89f6e2478f - std::panicking::rust_panic_with_hook::hc58e08e312fd25fa
                               at src/libstd/panicking.rs:474
   5:     0x7f89f6e24311 - std::panicking::continue_panic_fmt::h4c73c6c4fe871c84
                               at src/libstd/panicking.rs:381
   6:     0x7f89f6e2425e - std::panicking::begin_panic_fmt::h353e452ca5b41821
                               at src/libstd/panicking.rs:336
   7:     0x55b17d2063a3 - pulldown_cmark::parse::FirstPass::scan_containers::h42227ca29c3d9ac5
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:1019
   8:     0x55b17d2054f8 - pulldown_cmark::parse::FirstPass::parse_line::h8dfb7f768c4ec751
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:588
   9:     0x55b17d2006bb - pulldown_cmark::parse::FirstPass::parse_block::ha6a8cfc69b37e908
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:1117
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:384
  10:     0x55b17d20780a - pulldown_cmark::parse::Parser::new_with_broken_link_callback::h55e3f1d28ef432bb
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:243
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.4.1/src/parse.rs:2260
  11:     0x55b17cfe867d - rustdoc::html::markdown::markdown_links::h23f6a0144ed013ab
                               at src/librustdoc/html/markdown.rs:858
  12:     0x55b17d1ed0ed - <rustdoc::passes::collect_intra_doc_links::LinkCollector as rustdoc::fold::DocFolder>::fold_item::hbea0411aa3b6f869
                               at src/librustdoc/passes/collect_intra_doc_links.rs:271
  13:     0x55b17cf6df44 - <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter::hc88f1e03b971e4c9
                               at src/librustdoc/fold.rs:100
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/adapters/mod.rs:826
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/traits/iterator.rs:1572
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/adapters/mod.rs:826
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/traits/iterator.rs:1609
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/adapters/mod.rs:812
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1929
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1826
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1821
  14:     0x55b17d1e31b4 - rustdoc::fold::DocFolder::fold_inner_recur::hf136e342e77c4949
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1721
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/traits/iterator.rs:1465
                               at src/librustdoc/fold.rs:100
                               at src/librustdoc/fold.rs:27
  15:     0x55b17d1e3f3b - rustdoc::fold::DocFolder::fold_item_recur::h7a50d20024cc0902
                               at src/librustdoc/fold.rs:89
  16:     0x55b17d1efd8c - <rustdoc::passes::collect_intra_doc_links::LinkCollector as rustdoc::fold::DocFolder>::fold_item::hbea0411aa3b6f869
                               at src/librustdoc/passes/collect_intra_doc_links.rs:401
  17:     0x55b17cf6df44 - <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter::hc88f1e03b971e4c9
                               at src/librustdoc/fold.rs:100
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/adapters/mod.rs:826
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/traits/iterator.rs:1572
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/adapters/mod.rs:826
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/traits/iterator.rs:1609
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/adapters/mod.rs:812
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1929
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1826
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1821
  18:     0x55b17d1e31b4 - rustdoc::fold::DocFolder::fold_inner_recur::hf136e342e77c4949
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/vec.rs:1721
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/iter/traits/iterator.rs:1465
                               at src/librustdoc/fold.rs:100
                               at src/librustdoc/fold.rs:27
  19:     0x55b17d1e3f85 - rustdoc::fold::DocFolder::fold_item_recur::h7a50d20024cc0902
                               at src/librustdoc/fold.rs:90
  20:     0x55b17d1efd8c - <rustdoc::passes::collect_intra_doc_links::LinkCollector as rustdoc::fold::DocFolder>::fold_item::hbea0411aa3b6f869
                               at src/librustdoc/passes/collect_intra_doc_links.rs:401
  21:     0x55b17d1f0808 - <rustdoc::passes::collect_intra_doc_links::LinkCollector as rustdoc::fold::DocFolder>::fold_crate::ha8500a607e9b12ed
                               at src/librustdoc/passes/collect_intra_doc_links.rs:415
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/option.rs:624
                               at src/librustdoc/passes/collect_intra_doc_links.rs:415
  22:     0x55b17d1ea77a - rustdoc::passes::collect_intra_doc_links::collect_intra_doc_links::hb26917f271b5ea74
                               at src/librustdoc/passes/collect_intra_doc_links.rs:35
  23:     0x55b17d0911f1 - rustc::ty::context::tls::enter_global::h3ea23f6a8fdb614e
                               at src/librustdoc/core.rs:469
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/passes.rs:804
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1973
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1940
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1873
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1939
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1972
  24:     0x55b17d0e9684 - rustc_interface::passes::BoxedGlobalCtxt::access::{{closure}}::h2f7e2355c9f75881
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/passes.rs:804
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c//<::rustc_data_structures::box_region::declare_box_region_type macros>:17
  25:     0x7f89fa4f5754 - rustc_interface::passes::create_global_ctxt::{{closure}}::h23a9be9fc4f389fa
  26:     0x55b17d0e9144 - rustc_interface::passes::BoxedGlobalCtxt::enter::h0cc5282193701558
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/boxed.rs:910
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_data_structures/box_region.rs:52
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c//<::rustc_data_structures::box_region::declare_box_region_type macros>:19
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/passes.rs:804
  27:     0x55b17d0b3cca - rustc_interface::interface::run_compiler_in_existing_thread_pool::hd1082c23acf0e8dc
                               at src/librustdoc/core.rs:358
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/interface.rs:122
  28:     0x55b17d1a5374 - rustdoc::core::run_core::hccefa9c8e9602c53
                               at src/librustdoc/core.rs:344
  29:     0x55b17d0758b8 - std::panicking::try::do_call::hebc4c0f59cbfbac5
                               at src/librustdoc/lib.rs:450
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/panic.rs:309
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/panicking.rs:293
  30:     0x7f89f6e35649 - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:85
  31:     0x55b17d1afc02 - rustc_driver::report_ices_to_stderr_if_any::h5b8c669d9ae1dbe5
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/panicking.rs:272
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/panic.rs:388
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_driver/lib.rs:1114
  32:     0x55b17cf333f2 - rustdoc::main_args::h562be9f362b6e504
                               at src/librustdoc/lib.rs:447
                               at src/librustdoc/lib.rs:404
  33:     0x55b17d183bc2 - std::thread::local::LocalKey<T>::with::h48ab247e0351d650
                               at src/librustdoc/lib.rs:98
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/option.rs:416
                               at src/librustdoc/lib.rs:98
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/util.rs:186
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1928
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/thread/local.rs:299
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/thread/local.rs:245
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1920
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/thread/local.rs:299
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/thread/local.rs:245
  34:     0x55b17d0af0dd - scoped_tls::ScopedKey<T>::set::h0fbef2ebe7e1f2e9
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc/ty/context.rs:1912
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/util.rs:186
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/util.rs:182
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libsyntax/lib.rs:101
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137
  35:     0x55b17d0fe2a8 - syntax::with_globals::ha5f0305c0ff338f2
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libsyntax/lib.rs:100
  36:     0x55b17d15614a - std::sys_common::backtrace::__rust_begin_short_backtrace::h2b5109f02111a894
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/util.rs:181
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/librustc_interface/util.rs:159
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/sys_common/backtrace.rs:136
  37:     0x7f89f6e35649 - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:85
  38:     0x55b17d1b7a18 - core::ops::function::FnOnce::call_once{{vtable.shim}}::ha0cb8054e8801069
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/panicking.rs:272
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/panic.rs:388
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libstd/thread/mod.rs:468
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/libcore/ops/function.rs:231
  39:     0x7f89f6e0678e - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hf2a64fd13cf7aa92
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/boxed.rs:704
  40:     0x7f89f6e342bf - std::sys::unix::thread::Thread::new::thread_start::h6ac210c3e2a30669
                               at /rustc/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/src/liballoc/boxed.rs:704
                               at src/libstd/sys_common/thread.rs:13
                               at src/libstd/sys/unix/thread.rs:79
  41:     0x7f89f699b6da - start_thread
  42:     0x7f89f64ac88e - __clone
  43:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.36.0-nightly (9b67bd42b 2019-05-01) running on x86_64-unknown-linux-gnu

error: Could not document `font8x8`.

Caused by:
  process didn't exit successfully: `rustdoc --crate-name font8x8 src/lib.rs --color always -o /home/richard/src/font8x8-rs/target/doc --cfg 'feature="default"' --cfg 'feature="std"' --cfg 'feature="unicode"' -L dependency=/home/richard/src/font8x8-rs/target/debug/deps` (exit code: 1)
```

### PR DESCRIPTION
**Pull request by euclio**
Fixes #60482.

### CODE REVIEW

File: src/librustdoc/html/markdown.rs [hunk 1]
```diff
@@ -782,9 +783,8 @@ pub fn plain_summary_line_full(md: &str, limit_length: bool) -> String {
             let next_event = next_event.unwrap();
             let (ret, is_in) = match next_event {
                 Event::Start(Tag::Paragraph) => (None, 1),
-                Event::Start(Tag::Code) => (Some("`".to_owned()), 1),
-                Event::End(Tag::Code) => (Some("`".to_owned()), -1),
                 Event::Start(Tag::Header(_)) => (None, 1),
+                Event::Code(code) => (Some(format!("`{}`", code)), 0),
```
Comment (euclio) [inline src/librustdoc/html/markdown.rs]: This might need a `self.is_in` check like the Event::Text arm.

**Comment by euclio** [comment]
We may want to hold off on this until https://github.com/raphlinus/pulldown-cmark/pull/325 is merged.

**Comment by marcusklaas** [comment]
The pull request has been merged and we're now preparing a bug fixing release (0.5.2). Shouldn't be more than a day or two.

**Comment by marcusklaas** [comment]
0.5.2 has been released!

**Comment by GuillaumeGomez** [comment]
@euclio You can update I guess? ;)

**Comment by euclio** [comment]
@GuillaumeGomez Updated.

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+

**Comment by jonas-schievink** [comment]
This also affects beta (and shows up a lot in the crater runs in https://github.com/rust-lang/rust/issues/61401), nominating for backport

**Comment by QuietMisdreavus** [comment]
Accepting backport nomination.

**Pull request merged**